<launch>

  <arg name="use_right_arm" default="true" />
  <arg name="use_left_arm" default="true" />

  <arg if="$(arg use_right_arm)"      name="right_arm_text" value="right" />
  <arg unless="$(arg use_right_arm)"  name="right_arm_text" value="" />
  <arg if="$(arg use_left_arm)"       name="left_arm_text" value="left" />
  <arg unless="$(arg use_left_arm)"   name="left_arm_text" value="" />
  <arg name="arm_config" value="$(arg right_arm_text)_$(arg left_arm_text)" />

  <include file="$(find pr2_machine)/$(env ROBOT).machine" />

  <!--node pkg="point_cloud_converter" name="point_cloud_converter_preself" type="point_cloud_converter">
        <remap from="/points_in" to="/swissranger/scan" />
        <remap from="/points2_out" to="/swissranger/pointcloud2_raw" />
  </node-->
  
  <param name="robot_description" command="$(find xacro)/xacro.py '$(find pr2_gazebo)/robots/pr2_with_kinect.urdf.xacro'" />
  
  <!-- pr2 self filter kinect data -->
  <node pkg="robot_self_filter" type="self_filter" respawn="true" name="pr2_self_filter" output="screen">
     <!--remap from="cloud_in" to="/swissranger/pointcloud2_raw" /-->
     <remap from="cloud_in" to="/kinect_camera/points" />
     <remap from="cloud_out" to="/pointcloud2_pr2_self_filtered" />    
     <param name="sensor_frame" type="string" value="/head_kinect_optical_frame" />         
     <param name="subsample_value" type="double" value=".01"/>
     <rosparam command="load" file="$(find pr2_arm_navigation_perception)/config/self_filter_$(arg arm_config).yaml" />
     <!--rosparam command="load" file="$(find pr2_arm_navigation_perception)/config/self_filter.yaml" /-->
  </node>

  <param name="human_description" command="$(find xacro)/xacro.py '$(find al_robots)/defs/human.urdf.xacro'" />

  <!-- human self filter kinect data -->
  <node pkg="robot_self_filter" type="self_filter" respawn="true" name="human_self_filter" output="screen">
     <!--remap from="cloud_in" to="/swissranger/pointcloud2_raw" /-->
     <remap from="cloud_in" to="/pointcloud2_pr2_self_filtered" />
     <remap from="cloud_out" to="/pointcloud2_self_filtered" />
     <emap from="robot_description" to="human_description" />
     <param name="sensor_frame" type="string" value="/head_kinect_optical_frame" />         
     <param name="subsample_value" type="double" value=".01"/>
     <rosparam command="load" file="$(find al_robots)/conf/human_self_filter.yaml" />
     <!--rosparam command="load" file="$(find pr2_arm_navigation_perception)/config/self_filter.yaml" /-->
  </node>
  
  <!-- TODO: /pointcloud_filtered should be given as a parameter to tabletop_segmentation after the yaml file is processed by al_perception -->
  <include file="$(find tabletop_object_detector)/launch/tabletop_segmentation.launch">
    <arg name="tabletop_segmentation_points_in" value="/pointcloud_filtered" />
  <param name="tabletop_segmentation/z_filter_min" type="double" value="0.50" />
  <param name="tabletop_segmentation/z_filter_max" type="double" value="1.3" />
  <param name="tabletop_segmentation/y_filter_min" type="double" value="-1.2" />
  <param name="tabletop_segmentation/y_filter_max" type="double" value="1.2" />
  <param name="tabletop_segmentation/x_filter_min" type="double" value="0.3" />
  <param name="tabletop_segmentation/x_filter_max" type="double" value="1.5" />
  </include>
  
  <!-- cluster bounding box finder (required for ...)--> 
  <node name="cluster_bounding_box_finder" pkg="object_manipulator" type="cluster_bounding_box_finder_server.py" output="screen" respawn="false" />
  
	<rosparam command="load" file="$(find al_perception)/config/perception_settings.yaml" />
	<node name="al_perception" pkg="al_perception" type="al_perception" output="screen" respawn="false"/>

</launch>
